### PR TITLE
Remove usage of `QTextCodec` in Hydrogen Import plugin

### PR DESCRIPTION
**Pull request by messmerd**
This takes us one step closer to Qt 6 support (#6614) since `QTextCodec` is one of the classes removed in Qt 6 and keeping it would require the use of the `Qt5Compat` module, which we'd like to avoid.

`QTextCodec` was only used by the HydrogenImport plugin when importing old Hydrogen files that were saved using TinyXML before it supported UTF-8. HydrogenImport would use `QTextCodec` to try to get the current encoding from the locale, and then use that as a best guess for interpreting the XML data in the unspecified encoding it was saved in. None of this was ever reliable, since the encoding of the computer that saved the Hydrogen file might not be the same as the computer running LMMS and importing that file.

There is no good solution here, so I decided to simply assume the old Hydrogen files are UTF-8 encoded. The worst that can happen is someone's ancient Hydrogen files containing non-ASCII text of some random encoding becomes mojibake'd after importing into LMMS, which is something that already could have happened.

I also cleaned up the Hydrogen Import source files a little bit.

### CODE REVIEW

**Comment by Rossmaxx** [comment]
> See the commit messages for details.

You could copy the message here for easier access. 

Also, one thing i would suggest is that you open this PR against qt6 branch instead of master but doesn't really matter as we can merge master back in anyway.

File: plugins/HydrogenImport/LocalFileMng.h
Comment (Rossmaxx) [inline plugins/HydrogenImport/LocalFileMng.h:19]: Is this an unused variable?

**Comment by Rossmaxx** [comment]
Forgot to ask, have you built with qt6 to check if it works?

File: plugins/HydrogenImport/LocalFileMng.cpp
Comment (tresf) [inline plugins/HydrogenImport/LocalFileMng.cpp:199]: I'm not sure how common it is for this XML to be an encoding other than UTF8, so I don't know the severity of this change, however from what I'm reading, it's pretty bleak.  Newer versions of Qt6 will have ICU (International Components for Unicode) support, but this comment suggests that ICU support won't be landing on Windows anytime soon.  https://forum.qt.io/topic/138790/replacing-calls-to-qtextstream-setcodec-in-qt-6/11?_=1729692809826&lang=en-US

We may -- if we decide to, add a comment that references the upstream feature incase this UTF-8 assumption causes an unobvious code regression.

```cpp
// Qt 6.4+ (https://codereview.qt-project.org/c/qt/qtbase/+/393373)
QStringConverter(...) ...
```

Comment (tresf) [reply plugins/HydrogenImport/LocalFileMng.cpp:199]: What should also be noted is that `QString enc = QTextCodec::codecForLocale()->name();` seems to make an odd assumption that the current encoding of the current locale is suitable for parsing an XML, which seems -- at least at a glance -- an odd and bug-prone assumption.  Hard-coding UTF-8 seems more predictable, even if it causes some edge-case regressions where the system local and the XML are both coincidentally in a non-UTF8 encoding.

Comment (messmerd) [reply plugins/HydrogenImport/LocalFileMng.cpp:199]: You are correct. As I mention in the commit message, the original choice to use the "best match" encoding from the locale is error prone because there is no guarantee that the encoding used to save the file will be the same encoding of the user importing it in LMMS.

There is no good solution here, so I decided to simply assume the old Hydrogen files are UTF-8 encoded. The worst that can happen is someone's ancient Hydrogen files containing non-ASCII text of some random encoding becomes mojibake'd after importing into LMMS, which is something that already could have happened.

File: plugins/HydrogenImport/LocalFileMng.h
Comment (messmerd) [reply plugins/HydrogenImport/LocalFileMng.h:19]: Yeah, this is only ever used as a static class.

**Comment by messmerd** [comment]
@Rossmaxx 
> Also, one thing i would suggest is that you open this PR against qt6 branch instead of master but doesn't really matter as we can merge master back in anyway.

I'd rather that the master branch have these changes now than wait until the Qt 6 branch is merged

> Also might wanna look into the codefactor warnings.

Looks unrelated to anything I did

> Forgot to ask, have you built with qt6 to check if it works?

No, but it builds with Qt 5 and the component that Qt 6 removed is no longer there, so it should be fine

**Comment by Rossmaxx** [comment]
Ok then, would be better to merge master on qt6 once this is merged, then to remove the compat module from there.

**Review by tresf — APPROVED** [review]

File: plugins/HydrogenImport/LocalFileMng.cpp
Comment (tresf) [reply plugins/HydrogenImport/LocalFileMng.cpp:199]: Thanks.  Yes, I see how that the original PR already uncovered this. Sorry for any redundancies that it caused.

**Review by michaelgregorius — APPROVED** [review]
Looks good to me.

One additional option might be to point people to external and online file converters in the documentation, for example:
* Linux: `iconv` or `recode`
* Windows: `gc` (PowerShell)
* Online: https://www.freeformatter.com/convert-file-encoding.html

That way we put the "burden" of the conversion away from LMMS (which is accomplished with this PR) and to better suited external tools.

Another option might be to point people to these tools inside of the application in case of a failure to import the Hydrogen file.

Although it is a question anyway how often people will really run into these problems. If I understand correctly a user will have to import a very old Hydrogen file into an LMMS version which was built after this PR got merged. Sounds rather unlikely.

**Pull request merged**
